docs/cookbook/html-conversion.md

```diff
@@ -0,0 +1,980 @@
+          }
+        }
+      }
+    },
+    group: {
+      basics: {
+        label: 'Basics',
+        fields: [ 'title', 'headerImage', 'heading', 'subheading', 'author', 'publicationDate', 'main' ]
+      }
+    }
+  }
+};
+
+```
+
+<template v-slot:caption>
+modules/blog/index.js
+</template>
+</AposCodeBlock>
+
+The `index.js` file for the `blog-page` will also be quite similar. Again, it will have a header image, heading, and subheading.
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    perPage: 5
+  },
+  fields: {
+    add: {
+      headerImage: {
+        type: 'area',
+        label: 'Header image',
+        required: true,
+        options: {
+          widgets: {
+            '@apostrophecms/image': {}
+          },
+          max: 1
+        }
+      },
+      heading: {
+        type: 'string',
+        label: 'Heading'
+      },
+      subheading: {
+        type: 'string',
+        label: 'Subheading'
+      }
+    },
+    group: {
+      basics: {
+        label: 'Basics',
+        fields: [
+          'title',
+          'headerImage',
+          'heading',
+          'subheading'
+        ]
+      }
+    }
+  }
+};
+
+```
+
+<template v-slot:caption>
+modules/blog-page/index.js
+</template>
+</AposCodeBlock>
+
+In the code above we are adding a `perPage` option of `5`. This will limit the number of blog articles shown on the "Home" page. This can be adjusted to best serve your needs.
+
+### The blog index.html page
+
+The "Home" page of the template is essentially an `index.html` page that lists all of the blog articles. Just like with the default page, we are going to add our header to a `beforeMain` block. Within the `main` block, we will copy the `Main Content` section from the `index.html` HTML template. To convert it to dynamically show all of the blog articles from our site we will delete all of the code in each of the `Post preview` sections except the first. Then, we will wrap the first `Post preview` section in a `for` loop. Finally, we will modify the `Pager` section to show both newer and older posts.
+
+<AposCodeBlock>
+::: v-pre
+
+```django
+{% extends 'layout.html' %}
+
+{% import 'fragments/header.html' as header %}
+
+{% block beforeMain %}
+  {{ super() }}
+  {% render header.headerArea(data.page) %}
+{% endblock %}
+
+{% block main %}
+  <!-- Main Content-->
+  <div class="container px-4 px-lg-5">
+    <div class="row gx-4 gx-lg-5 justify-content-center">
+      <div class="col-md-10 col-lg-8 col-xl-7">
+        {% for piece in data.pieces %}
+          <!-- Post Preview -->
+          <div class="post-preview">
+            <a href="{{ piece._url }}">
+              <h2 class="post-title">{{ piece.heading }}</h2>
+              <h3 class="post-subtitle">{{ piece.subheading }}</h3>
+            </a>
+            <p class="post-meta"> Posted by {{ piece.author }} on {{ piece.publicationDate | date('MMMM d, YYYY') }}</p>
```

Comment:
Most of the fields in the schema aren't required for the header or the piece type. Should more of them be required with conditionals in the template for the one's that aren't?

Reply:
Yeah - didn't think that through! I went back and made all the fields required.